### PR TITLE
feat: `Series.pipe` and `Expr.pipe`

### LINKED ISSUE #574: feat: Add `Series.pipe`, similar to `DataFrame.pipe`
Take a look at `DataFrame.pipe`, this should be quite similar

### PR DESCRIPTION
**Pull request by muddi900**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #574
- Closes #574

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

**Review by FBruzzesi — COMMENTED** [review]
Hey there! Thanks for you PR! This looks promising already, yet could you make sure to add dedicated unit tests in the `/tests/expr_and_series` folder?

**Comment by muddi900** [comment]
So the series test seem have the df as default constructor value. How would that test Series methods?

**Comment by MarcoGorelli** [comment]
thanks for your PR! you can use `constructor_series`

**Comment by muddi900** [comment]
My series test keeps failing at pyarrow

```
AttributeError: 'ArrowSeries' object has no attribute '__native_namespace__'. Did you mean: '__narwhals_namespace__'?
```

**Comment by MarcoGorelli** [comment]
ah we just added that - if you fetch and merge `upstream/main`, I think it should pass

**Comment by muddi900** [comment]
Another issue with pyarrow:

```AttributeError: 'pyarrow.lib.ChunkedArray' object has no attribute 'to_list'. Did you mean: 'to_pylist'?```

**Comment by MarcoGorelli** [comment]
> Another issue with pyarrow:
> AttributeError: 'pyarrow.lib.ChunkedArray' object has no attribute 'to_list'. Did you mean: 'to_pylist'?

where did you see this? could you add a test please? even if it fails, then it's easier to provide feedback

File: tests/expr_and_series/pipe_test.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from tests.utils import compare_dicts
+
+
+@nw.narwhalify()
+def func(s: Any) -> Any:
+    return s.pipe(lambda x: x**2)
+
+
+input_list = {"a": [2, 4, 6, 8]}
+expected = [i**2 for i in input_list["a"]]
```
Comment (MarcoGorelli) [inline tests/expr_and_series/pipe_test.py:13]: can you construct `expected` explicitly please? else we risk having a bug in both paths and it going unnoticed

**Comment by muddi900** [comment]
I got this error in test.

**Comment by MarcoGorelli** [comment]
which error?

**Comment by muddi900** [comment]
> which error?

AttributeError: 'pyarrow.lib.ChunkedArray' object has no attribute 'to_list'. Did you mean: 'to_pylist'?

File: tests/expr_and_series/pipe_test.py (hunk 1 above)
Comment (MarcoGorelli) [reply tests/expr_and_series/pipe_test.py:13]: this comment hasn't been addressed, please don't click 'resolve' until it has

File: tests/expr_and_series/pipe_test.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    df = nw.from_native(constructor(input_list))
+    e = df.select(nw.col("a").pipe(func))
+    compare_dicts(e, {"a": expected})
+
+
+def test_pipe_series(
+    constructor_series: Any,
+) -> None:
+    s = nw.from_native(constructor_series(input_list["a"]), series_only=True)
+    assert s.pipe(func).to_list() == expected
```
Comment (MarcoGorelli) [inline tests/expr_and_series/pipe_test.py:26]: `func` is a narwhalified function which uses `pipe`, and you're passing that to `pipe`? I'm confused sorry

Comment (muddi900) [reply tests/expr_and_series/pipe_test.py:26]: I am passing it to the pipe. See the docstring for the method: https://github.com/narwhals-dev/narwhals/blob/a210baf141ef413e3a913f03f69ebe45ad6c78e6/narwhals/series.py#L106-L138

File: tests/expr_and_series/pipe_test.py (hunk 1 above)
Comment (muddi900) [reply tests/expr_and_series/pipe_test.py:13]: I just haven't pushed the commit, but I am declaring it explicitly locally.

File: tests/expr_and_series/pipe_test.py (hunk 2 above)
Comment (muddi900) [reply tests/expr_and_series/pipe_test.py]: I fixed the tests. Thanks for pointing it out. However the ChunkedArray issue remains.

File: tests/expr_and_series/pipe_test.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    return s.pipe(lambda x: x**2)
+
+
+input_list = {"a": [2, 4, 6, 8]}
+expected = [4, 16, 36, 64]
+
+
+def test_pipe_expr(constructor: Any) -> None:
+    df = nw.from_native(constructor(input_list))
+    e = df.select(func(nw.col("a")))
```
Comment (MarcoGorelli) [inline tests/expr_and_series/pipe_test.py:18]: just use `nw.col('a').pipe(lambda x: x**2)`, no need for func

File: tests/expr_and_series/pipe_test.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    compare_dicts(e, {"a": expected})
+
+
+def test_pipe_series(
+    constructor_series: Any,
+) -> None:
+    s = nw.from_native(constructor_series(input_list["a"]), series_only=True)
+    assert func(s).to_list() == expected
```
Comment (MarcoGorelli) [inline tests/expr_and_series/pipe_test.py:26]: just do `s.pipe(lambda x: x**2)`

**Review by MarcoGorelli — COMMENTED** [review]
thanks for updating - please remove `def func` from the test, I think it's messing with other things (and not really necessary, your input is already narwhalified there)

**Comment by MarcoGorelli** [comment]
thanks for updating, can you check your changes please?

**Review by MarcoGorelli — APPROVED** [review]

**Pull request merged**